packages/nuxt/src/app/nuxt.ts

```diff
@@ -336,12 +358,37 @@ export function normalizePlugins (_plugins: Plugin[]) {
     console.warn('[warn] [nuxt] You are using a plugin that has not been wrapped in `defineNuxtPlugin`. It is advised to wrap your plugins as in the future this may enable enhancements:', unwrappedPlugins.map(p => p.name || p).join(','))
   }
 
-  return plugins as Plugin[]
+  return [
+    ...plugins.payload,
+    ...plugins.pre,
+    ...plugins.default,
+    ...plugins.post
+  ]
 }
 
-export function defineNuxtPlugin<T extends Record<string, unknown>> (plugin: Plugin<T>) {
-  plugin[NuxtPluginIndicator] = true
-  return plugin
+export function defineNuxtPlugin<T extends Record<string, unknown>> (plugin: Plugin<T> | ObjectPluginInput<T>) {
+  if (typeof plugin === 'function') {
+    plugin[NuxtPluginIndicator] = true
+    return plugin
```

Comment:
Maybe add a basic meta here too? (name could be infered but better, enforce explicitly added, etc)